src/tools/tidy/src/fluent_alphabetical.rs

```diff
@@ -13,11 +14,18 @@ fn filter_fluent(path: &Path) -> bool {
     if let Some(ext) = path.extension() { ext.to_str() != Some("ftl") } else { true }
 }
 
-fn check_alphabetic(filename: &str, fluent: &str, bad: &mut bool) {
+fn check_alphabetic(
+    filename: &str,
+    fluent: &str,
+    bad: &mut bool,
+    msgs: &mut HashMap<String, String>,
+) {
     let mut matches = MESSAGE.captures_iter(fluent).peekable();
     while let Some(m) = matches.next() {
+        let name = m.get(1).unwrap();
+        msgs.insert(name.as_str().to_owned(), filename.to_owned());
```

Comment:
What if the message exists already? This seems like a convenient place to sanity check that and error in that case.